OpenRA.Game/Actor.cs

```diff
@@ -320,6 +322,30 @@ public bool CanBeViewedByPlayer(Player player)
 			return defaultVisibility.IsVisible(this, player);
 		}
 
+		public IEnumerable<string> GetAllTargetTypes()
```

Comment:
On second thought, since I can't move it to ActorExts, I think I'll leave it for now.

I don't know if a region just for these will help. Maybe another PR can look into reorganising/refactoring the Actor class a bit.